File: day1/main.cpp
#include <iostream>
#include <fstream>
#include <string>

using namespace std;

int main() {
	ifstream file("input.txt");
	string line;

	int dial = 50;
	int count = 0;
	int count2 = 0;

	while (getline(file, line)) {
		int num = stoi(line.substr(1));
		switch (line[0]) {
			case 'L':
				for (int i = 0; i < num; i++) {
					dial -= 1;
					dial = dial % 100;
					if (dial == 0) {
						count2++;
					}
				}
				break;
			case 'R':
				for (int i = 0; i < num; i++) {
					dial += 1;
					dial = dial % 100;
					if (dial == 0) {
						count2++;
					}
				}
				break;
			default:
				cout << "bad line\n";
				return 1;
		}
		if (dial == 0) {
			count++;
		}
	}

	cout << "Part 1: " << count << endl;
	cout << "Part 2: " << count2 << endl;

	return 0;
}

File: day1/Main.java
import java.nio.file.Files;
import java.nio.file.Path;
import java.util.function.BiFunction;
import java.util.function.Function;
import java.util.stream.IntStream;
import java.util.stream.Stream;

void main() throws Exception {
	System.out.println("Part 1: " + ((BiFunction<Stream<String>, Integer[], Integer>) ((stream, dial) -> stream.mapToInt(line -> ((dial[0] += (Integer.parseInt(line.substring(1)) * (line.charAt(0) == 'L' ? -1 : 1))) % 100 == 0) ? 1 : 0).sum())).apply(Files.lines(Path.of("input.txt")), new Integer[]{50}));
	System.out.println("Part 2: " + ((BiFunction<Stream<String>, Integer[], Integer>) ((stream, dial) -> stream.mapToInt(line -> ((Function<Integer, Integer>) (d -> (int) (line.charAt(0) == 'L' ? IntStream.rangeClosed(d, d + Integer.parseInt(line.substring(1)) - 1) : IntStream.rangeClosed(d - Integer.parseInt(line.substring(1)) + 1, d)).filter(n -> n % 100 == 0).count())).apply(dial[0] += Integer.parseInt(line.substring(1)) * (line.charAt(0) == 'R' ? 1 : -1))).sum())).apply(Files.lines(Path.of("input.txt")), new Integer[]{50}));
}
